dbt/include/snowflake/macros/materializations/table.sql

```diff
@@ -85,3 +84,34 @@ def main(session):
 # dbt = dbtObj(session.table)
 # df = model(dbt, session)
 {%endmacro%}
+
+
+{% macro drop_old_relation_as_needed(old_relation, target_relation) %}
```

Comment:
Could we move this logic into python? It's probably easier as we could write this comparison logic into the Relation class

Reply:
Agreed, something like this would read nicely:

```sql
{% if target_relation.needs_to_drop(old_relation) %}
    {{ drop_relation_if_exists(old_relation) }}
{% endif %}
```

Reply:
Thirded, this is great!